### PR TITLE
chore(Jest): top level spec [PART-5]

### PR DESCRIPTION
**Pull request by samouss**
This PR migrate the files `test/spec/*` to Jest. It also moves the mocked data `search.testdata` outside of the `spec` folder. It avoids that Jest runs it. You can run the test locally with:

```sh
yarn jest test/spec
```

The migration has been done by [`jest-codemods`](https://github.com/skovhus/jest-codemods) (recommended by the [Jest](https://jestjs.io/docs/en/migration-guide#jest-codemods) project) for all the assertions. Then I removed the dependency on `sinon` file by file. I tried to didn't update the code style or anything unless it was too complicated to migrate.

### CODE REVIEW

File: test/spec/refinements.js [hunk 1]
```diff
@@ -1,95 +1,85 @@
 'use strict';
 
-var test = require('tape');
 var _ = require('lodash');
 var algoliasearchHelper = require('../../index');
 
 var emptyClient = {};
 
-test('Adding refinments should add an entry to the refinments attribute', function(t) {
+test('Adding refinments should add an entry to the refinments attribute', function() {
```
Comment (francoischalifour) [inline test/spec/refinements.js]:
```suggestion
test('Adding refinements should add an entry to the refinements attribute', function() {
```

File: test/spec/refinements.js [hunk 2]
```diff
@@ -1,95 +1,85 @@
...
   var facetName = 'facet1';
   var facetValue = '42';
 
   var helper = algoliasearchHelper(emptyClient, 'index', {
     facets: [facetName]
   });
 
-  t.ok(_.isEmpty(helper.state.facetsRefinements), 'should be empty at first');
+  expect(_.isEmpty(helper.state.facetsRefinements)).toBeTruthy();
```
Comment (francoischalifour) [inline test/spec/refinements.js]: Can be stricter.

File: test/spec/refinements.js [hunk 3]
```diff
@@ -1,95 +1,85 @@
...
   helper.addRefine(facetName, facetValue);
-  t.ok(_.size(helper.state.facetsRefinements) === 1, 'when adding a refinment, should have one');
-  t.deepEqual(helper.state.facetsRefinements.facet1, [facetValue]);
+  expect(_.size(helper.state.facetsRefinements) === 1).toBeTruthy();
```
Comment (francoischalifour) [inline test/spec/refinements.js]: Can use Jest's `toHaveLength`.

File: test/spec/refinements.js [hunk 4]
```diff
@@ -1,95 +1,85 @@
...
-test('Adding several refinements for a single attribute should be handled', function(t) {
+test('Adding several refinements for a single attribute should be handled', function() {
   var facetName = 'facet';
 
   var helper = algoliasearchHelper(emptyClient, null, {
     facets: [facetName]
   });
 
-  t.ok(_.isEmpty(helper.state.facetsRefinements), 'empty');
+  expect(_.isEmpty(helper.state.facetsRefinements)).toBeTruthy();
```
Comment (francoischalifour) [inline test/spec/refinements.js]: Can use Jest's `toHaveLength`.

File: test/spec/refinements.js [hunk 5]
```diff
@@ -98,58 +88,50 @@ test('isDisjunctiveRefined', function(t) {
...
-  t.equal(helper.isRefined('facet1', 'boom'), true, 'the facet + value is refined >> true');
-
-  t.equal(helper.isRefined('facet1', 'booohh'), false, 'value not refined but is a facet');
-  t.throws(_.bind(helper.isRefined, helper, 'notAFacet', 'maoooh'), 'should throw as it is not a facet');
-  t.throws(_.bind(helper.isRefined, helper, null, null), 'not valid values');
+  expect(helper.isRefined('facet1', 'boom')).toBe(true);
 
-  t.end();
+  expect(helper.isRefined('facet1', 'booohh')).toBe(false);
+  expect(_.bind(helper.isRefined, helper, 'notAFacet', 'maoooh')).toThrow();
```
Comment (francoischalifour) [inline test/spec/refinements.js]: Do we want to track the errors?

File: test/spec/requestBuilder.js [hunk 6]
```diff
@@ -1,59 +1,48 @@
...
   var searchParams = testData.searchParams;
 
   var queries = getQueries(searchParams.index, searchParams);
-  t.equal(queries.length, 2);
-  t.equal(queries[0].params.clickAnalytics, undefined, 'the parameter clickAnalytics should be defined on the first query');
-  t.equal(queries[1].params.clickAnalytics, false, 'the parameter clickAnalytics should be set to false on the second query');
-
-  t.end();
+  expect(queries.length).toBe(2);
+  expect(queries[0].params.clickAnalytics).toBe(undefined);
```
Comment (francoischalifour) [inline test/spec/requestBuilder.js]: Can use `toBeUndefined()`.

File: test/spec/refinements.js (hunk 2 above)
Comment (samouss) [reply test/spec/refinements.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/refinements.js (hunk 3 above)
Comment (samouss) [reply test/spec/refinements.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/refinements.js (hunk 4 above)
Comment (samouss) [reply test/spec/refinements.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/refinements.js (hunk 5 above)
Comment (samouss) [reply test/spec/refinements.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/requestBuilder.js (hunk 6 above)
Comment (samouss) [reply test/spec/requestBuilder.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

**Pull request merged**